### PR TITLE
Removed Spatial Enrichment + Cython

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Removes Cython related files and Spatial Enrichment functionality from the repo. Updates the build process to reflect that. Using `uv` instead of regular python GitHub Actions. Updated the README badges.

**How did you implement your changes**

- Removed `_bootstrapping.pyx` and `setup.py` files.
- Removed `spatial_enrichment` files.
- Updated GitHub Actions workflows to use `uv` for building, testing and publishing. Removed `cibuldwheel` components of the workflows. 
  - Due to ark now being a pure python package we do not need to build nor test on multiple platforms, so defaulted to `ubuntu-latest` for builds and tests. This simplifies much of the CI workflow and speeds it up. It is a good practice to still test across various Python versions however.
- Freshened up the README badges with some new ones.

**Remaining issues**
N/A.

### CODE REVIEW

File: src/ark/analysis/spatial_analysis_utils.py [hunk 1]
```diff
@@ -275,78 +274,6 @@ def compute_close_cell_num(dist_mat, dist_lim, analysis_type,
...
-
-    # unpermute close_num_rand
-    x_scramble = np.tile(np.argsort(sort_permutation), (len(sort_permutation), 1))
-    y_scramble = x_scramble.T
-    close_num_rand = close_num_rand[x_scramble, y_scramble, :]
-
-    return close_num_rand
-
-
 def calculate_enrichment_stats(close_num, close_num_rand):
```
Comment (camisowers) [inline src/ark/analysis/spatial_analysis_utils.py:277]: I think this function can be deleted as well.

File: tests/analysis/spatial_analysis_utils_test.py [hunk 2]
```diff
@@ -199,35 +199,6 @@ def test_compute_close_cell_num():
...
-    # bad marker nums
-    marker_nums[0] = example_distmat.shape[0] + 1
-
-    with pytest.raises(ValueError):
-        example_closenumrand = spatial_analysis_utils.compute_close_cell_num_random(
-            marker_nums, marker_pos_labels, example_distmat, dist_lim=100, bootstrap_num=100
-        )
-
-
 def test_calculate_enrichment_stats():
```
Comment (camisowers) [inline tests/analysis/spatial_analysis_utils_test.py:202]: This too

**Review by camisowers — CHANGES_REQUESTED** [review]
Few additional removals. Also seems like the builds/tests aren't running.

**Review by camisowers — APPROVED** [review]
Looks good!

**Review by alex-l-kong — APPROVED** [review]

**Pull request merged**